src/main.rs

```diff
@@ -96,7 +96,7 @@ enum Commands {
     /// Create a basin.
     CreateBasin {
         /// Name of the basin to create.
-        basin: BasinNameOnlyUri,
```

Comment:
this does not work apparently :o

Reply:
Work in what way?

Reply:
it doesnt render out the miette diagnostic! not sure why

Reply:
It's because it's a clap parse error

Reply:
yup I assumed clap won't show that  after trying a few hacks, feel free to push here in case you find a better way

Reply:
I don't think there's a better way, but, we can assume in this case, it isn't a valid basin name. `None` scheme basically means that the input is at-max a valid file path which doesn't strongly imply that the user tried to input a URI. We can just return the parse basin error in this case?

```diff
diff --git a/src/types.rs b/src/types.rs
index 14d931a..cc249d4 100644
--- a/src/types.rs
+++ b/src/types.rs
@@ -33,7 +33,7 @@ fn parse_maybe_basin_or_uri(
         }
         Err(parse_basin_err) => {
             // Should definitely be a URI else error.
-            let uri = http::Uri::from_str(s).map_err(|_| parse_basin_err)?;
+            let uri = http::Uri::from_str(s).map_err(|_| parse_basin_err.clone())?;

             match uri.scheme_str() {
                 Some("s2") => (),
@@ -45,10 +45,9 @@ fn parse_maybe_basin_or_uri(
                     )));
                 }
                 None => {
-                    return Err(BasinNameOrUriParseError::InvalidUri(miette::miette!(
-                        help = "Make sure the URI starts with 's2://'",
-                        "Missing URI scheme"
-                    )))
+                    // It's probably not an attempt to enter a URI. Safe to
+                    // assume this is simply an invalid basin.
+                    return Err(parse_basin_err.into());
                 }
             };
```

Reply:
As far as trying to return error with help message goes, we'll just have to format the error ourselves or do what you did.

What you did in this PR isn't the worst option. I would, though, prefer we do it consistently for all the commands.